golang/cosmos/x/vtransfer/keeper/keeper.go

```diff
@@ -101,12 +102,42 @@ func (k Keeper) GetReceiverImpl() vibctypes.ReceiverImpl {
 	return k
 }
 
+// Extract the base address from the packet sender (if senderIsLocal) or
+// receiver (if !senderIsLocal), since the local ibcModule doesn't understand
+// address parameters.
+func (k Keeper) packetWithOnlyBaseAddresses(packet channeltypes.Packet, senderIsLocal bool) channeltypes.Packet {
+	transferData := transfertypes.FungibleTokenPacketData{}
+	if err := k.cdc.UnmarshalJSON(packet.GetData(), &transferData); err != nil {
+		return packet
+	}
+	if senderIsLocal {
+		baseSender, err := types.ExtractBaseAddress(transferData.Sender)
+		if err == nil {
+			transferData.Sender = baseSender
+		}
+	} else {
+		baseReceiver, err := types.ExtractBaseAddress(transferData.Receiver)
+		if err == nil {
+			transferData.Receiver = baseReceiver
+		}
+	}
+
+	// Re-serialize the packet data with the base addresses.
+	data, _ := k.cdc.MarshalJSON(&transferData)
+
+	// Replace the packet's data with the new data.
+	packet.Data = data
+	return packet
+}
```

Comment:
Is it expected behavior that when there is an error, we send an empty JSON on Data?

Reply:
Good catch!  No, not expected.  I'll fix this while working on its unit tests.